Plugin/Model/OrderItemRepositoryPlugin.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/*
+ * Copyright Extend (c) 2023. All rights reserved.
+ * See Extend-COPYING.txt for license details.
+ */
+
+namespace Extend\Integration\Plugin\Model;
+
+use Magento\Sales\Api\OrderItemRepositoryInterface;
+use Magento\Sales\Api\Data\OrderItemSearchResultInterface;
+use Magento\Sales\Api\Data\OrderItemExtensionFactory;
+use Magento\Quote\Model\ResourceModel\Quote\Item\CollectionFactory;
+
+class OrderItemRepositoryPlugin
+{
+    /**
+     * @var OrderItemExtensionFactory
+     */
+    private OrderItemExtensionFactory $orderItemExtensionFactory;
+
+    /**
+     * @var CollectionFactory
+     */
+    private CollectionFactory $quoteItemCollectionFactory;
+
+    public function __construct(
+        OrderItemExtensionFactory $orderItemExtensionFactory,
+        CollectionFactory $quoteItemCollectionFactory
+    ) {
+        $this->orderItemExtensionFactory = $orderItemExtensionFactory;
+        $this->quoteItemCollectionFactory = $quoteItemCollectionFactory;
+    }
+
+    /**
+     * This plugin injects product protection product data into the order's product protection order items
+     *
+     * @param OrderItemRepositoryInterface $subject
+     * @param OrderItemSearchResultInterface $searchResult
+     * @return OrderItemSearchResultInterface
+     */
+    public function afterGetList(
+        OrderItemRepositoryInterface $subject,
+        OrderItemSearchResultInterface $searchResult
+    ): OrderItemSearchResultInterface {
+        $orderItems = $searchResult->getItems();
+
```

Comment:
Can we add an if statement here to ensure $orderItems exist and is a valid array?

Reply:
done 👍